commands/purge_config_command.go

```diff
@@ -69,3 +80,9 @@ func (c *PurgeConfigCommand) Execute(args []string) ExecutionStatus {
 func (c *PurgeConfigCommand) reportError(err error) {
 	ui.Failed("Could not purge configuration: %v\n", err)
 }
+
+// NewSessionProvider Returns a new SessionProvider - responponsible for giving a unique token each time
+func (c *PurgeConfigCommand) NewSessionProvider(host string) (csrf.SessionProvider, error) {
```

Comment:
What about the MtaClient? Will it be problem when you are working with the MtaClient?

Reply:
Now it works with RestClient. Do you mean to switch it to MtaClient?
I have tried to do so but session is kept for only one client

Reply:
I meant: Will it be a problem if I use the method and in the same way I am working with the MtaClient? What about the http calls there? Will they be able to execute POST calls when the MTA Client did not call the /api/v1/csrf-token endpoint?

Reply:
MTA Client will continue to call /api/v1/csrf-token so further POST calls are ok

Reply:
I understand now. Could we make the GetSession method in the MtaClient to call the endpoint twice(as is now in the REST client)?